src/test/java/subway/Line/LineAcceptanceTest.java

```diff
@@ -0,0 +1,258 @@
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.Sql.ExecutionPhase;
+
+@SuppressWarnings("NonAsciiCharacters")
+@DisplayName("지하철 노선 관련 기능")
+@Sql(scripts = {"/truncate.sql"}, executionPhase = ExecutionPhase.BEFORE_TEST_METHOD)
+@SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
+public class LineAcceptanceTest extends LineAcceptanceFixture {
+    Long 신사역_ID;
+    Long 강남역_ID;
+    Long 청량리_ID;
+    Long 서울숲_ID;
+
+    @BeforeEach
+    void setUp() {
+        신사역_ID = requestCreateStation("신사역")
+                .jsonPath()
+                .getObject("id", Long.class);
+
+        강남역_ID = requestCreateStation("강남역")
+                .jsonPath()
+                .getObject("id", Long.class);
+
+        청량리_ID = requestCreateStation("청량리")
+                .jsonPath()
+                .getObject("id", Long.class);
+        서울숲_ID = requestCreateStation("서울숲")
+                .jsonPath()
+                .getObject("id", Long.class);
+    }
+
+    /**
+     * Given: 새로운 지하철 노선 정보를 입력하고,
+     * When: 관리자가 노선을 생성하면,
+     * Then: 해당 노선이 생성되고 노선 목록에 포함된다.
+     */
+    @DisplayName("지하철 노선을 생성한다")
+    @Test
+    void createLine() {
+        // Given & When
+        Map<String, Object> 신분당선 = 신분당선_생성(신사역_ID, 강남역_ID);
+        ExtractableResponse<Response> response = requestCreateLine(신분당선);
+
+        // Then
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.CREATED.value());
+        List<Long> stationIds = response.jsonPath().getList("stations.id", Long.class);
+        assertThat(stationIds).containsExactlyInAnyOrder(신사역_ID, 강남역_ID);
+    }
+
+    /**
+     * Given: 여러 개의 지하철 노선이 등록되어 있고,
+     * When: 관리자가 지하철 노선 목록을 조회하면,
+     * Then: 모든 지하철 노선 목록이 반환된다.
+     */
+    @DisplayName("모든 지하철 노션을 목록을 조회한다.")
+    @Test
+    void readLines() {
+        // Given
+        Long 신분당선_ID = requestCreateLine(신분당선_생성(신사역_ID, 강남역_ID)).jsonPath().getObject("id", Long.class);
+        Long 분당선_ID = requestCreateLine(분당선_생성(청량리_ID, 서울숲_ID)).jsonPath().getObject("id", Long.class);
+
+        // When
+        ExtractableResponse<Response> response = RestAssured.given().log().all().when().get("/lines").then().log().all()
+                .extract();
+
+        // Then
+        List<Long> lineIds = response.jsonPath().getList("id", Long.class);
+        assertThat(lineIds).containsExactlyInAnyOrder(신분당선_ID, 분당선_ID);
+
+        List<List<Integer>> ids = response.jsonPath().getList("stations.id");
+        ids.get(0).addAll(ids.get(1));
+        List<Long> stationIds = ids.get(0).stream().map(Long::valueOf).collect(Collectors.toList());
+        assertThat(stationIds).containsExactlyInAnyOrder(신사역_ID, 강남역_ID, 청량리_ID, 서울숲_ID);
```

Comment:
```suggestion
        // Given
        var 신분당선_ID = requestCreateLine(신분당선_생성(신사역_ID, 강남역_ID)).jsonPath().getObject("id", Long.class);
        var 분당선_ID = requestCreateLine(분당선_생성(청량리_ID, 서울숲_ID)).jsonPath().getObject("id", Long.class);

        // When
        var response = RestAssured.given().log().all().when().get("/lines").then().log().all()
                .extract();

        // Then
        var lineIds = response.jsonPath().getList("id", Long.class);
        assertThat(lineIds).containsExactlyInAnyOrder(신분당선_ID, 분당선_ID);

        var ids = response.jsonPath().getList("stations.id");
        ids.get(0).addAll(ids.get(1));
        var = ids.get(0).stream().map(Long::valueOf).collect(Collectors.toList());
        assertThat(stationIds).containsExactlyInAnyOrder(신사역_ID, 강남역_ID, 청량리_ID, 서울숲_ID);
```

타입 추론을 사용하여 더 간결하게 만들어볼까요?